src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -218,13 +227,19 @@ private synchronized void updateSubscriptionsInternal() {
         Set<String> topicsToSubscribe = new HashSet<>(toSubscribeLocalMqttTopics);
         topicsToSubscribe.removeAll(subscribedLocalMqttTopics);
 
-        // TODO: Support configurable qos, add retry
+        // TODO: Support configurable qos
         topicsToSubscribe.forEach(s -> {
             try {
-                mqttClientInternal.subscribe(s);
+                RetryUtils.runWithRetry(mqttExceptionRetryConfig, () -> {
```

Comment:
what thread is running this? It will be blocking until success

Reply:
We're running entire subscribe workflow in a separate thread: [here](https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/1b9ca0f9e0a2fcad6d3d752605b19f281f9f2d53/src/main/java/com/aws/greengrass/mqttbridge/MessageBridge.java#L107-L108). It'll block this separate thread until success.